xernel/kernel/src/arch/amd64/mod.rs

```diff
@@ -61,6 +62,23 @@ pub fn read_cr2() -> VirtAddr {
     }
 }
 
+#[inline]
+pub fn read_cr8() -> IPL {
+    let value: u64;
+
+    unsafe {
+        asm!("mov {}, cr8", out(reg) value, options(nomem, nostack, preserves_flags));
+    }
+    IPL::from(value)
+}
+
+#[inline]
+pub fn write_cr8(ipl: IPL) {
```

Comment:
Maybe move this function libxernel, as it also exists in spin.rs and the same function is also needed in ipl.rs